File: 1478-maximum-number-of-events-that-can-be-attended/maximum-number-of-events-that-can-be-attended.cs
public class Solution {
    public int MaxEvents(int[][] events) {
        Array.Sort(events, (a, b) => a[0].CompareTo(b[0]));

        var minHeap = new PriorityQueue<int, int>(); 
        int day = 0, i = 0, res = 0, n = events.Length;

        while(i < n || minHeap.Count > 0)
        {
            if(minHeap.Count == 0)
            {
                day = events[i][0];
            }

            while(i < n && events[i][0] <= day)
            {
                minHeap.Enqueue(events[i][1], events[i][1]);
                i++;
            }

            while(minHeap.Count > 0 && minHeap.Peek() < day)
            {
                minHeap.Dequeue();

            }

            if(minHeap.Count > 0)
            {
                minHeap.Dequeue();
                res++;
                day++;
            }
        }

        return res;
    }
}

File: 1478-maximum-number-of-events-that-can-be-attended/maximum-number-of-events-that-can-be-attended.cpp
class Solution {
public:
    int maxEvents(vector<vector<int>>& events) {
        sort(events.begin(), events.end(), [](const vector<int>& a, const vector<int>& b) {
            return a[1] < b[1];
        });

        set<int> set;
        for(int i = 1;i <= 100000;i++) set.insert(i);
        int count = 0;

        for (const vector<int>& event : events) {
            int start = event[0];
            int end = event[1];

            auto lb = set.lower_bound(start);
            if(lb == set.end() || *lb > end)
            {
                continue;
            }
            else
            {
                count++;
                set.erase(*lb);
            }
        }

        return count;
    }
};